pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -138,17 +134,23 @@ def setup(self, model, max_batches, dataloaders):
         self.num_dataloaders = self._get_num_dataloaders(dataloaders)
 
     def on_evaluation_epoch_start(self, *args, **kwargs):
-        if self.testing:
+        if self.trainer.testing:
             self.trainer.call_hook('on_test_epoch_start', *args, **kwargs)
         else:
             self.trainer.call_hook('on_validation_epoch_start', *args, **kwargs)
 
-    def build_args(self, test_mode, batch, batch_idx, dataloader_idx):
+    def build_args(self, batch, batch_idx, dataloader_idx):
```

Comment:
this will be a breaking change, pls keep it and add a warning for now+v0.2 about deprecation

Reply:
isn't this an internal API change?

Reply:
if so let's make it as protected `_ build_args` so anyone using it will crash rather on the name than bad arguments which are harder to spot...